torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -257,9 +258,10 @@ void ProcessGroupAgent::enqueueRecv(RecvWork work) {
 
       Message message = deserialize(work.type_, ss);
 
-      if (message.isRequest()) {
-        auto response = cb_(std::move(message));
-        send(work.from_, std::move(response));
+      if (message.requiresResponse()) {
```

Comment:
I believe we should always send the response, at least as an acknowledgment, and the client should not throw away the acknowledgement, at least for tracing purposes. (e.g. if a FORK_DELETE times out, we may want to re-send the request. etc).

Reply:
Will address this in #25499